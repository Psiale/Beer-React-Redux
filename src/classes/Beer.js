const Beer = (id, name, tagline, description, imageUrl, foodPairing) => ({
  name, tagline, description, imageUrl, foodPairing,
});

export default Beer;
