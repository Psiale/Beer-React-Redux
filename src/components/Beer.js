/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';

const Beer = ({ beer, handleOnClick }) => {
  const {
    name, tagline, abv, description, image_url, food_pairing,
  } = beer;
  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    <div onClick={handleOnClick}>
      <div>{name}</div>
      <div>{tagline}</div>
      <div>{abv}</div>
      <div>{description}</div>
      <img src={image_url} alt="" />
      <div>
        {food_pairing}
      </div>
    </div>
  );
};

Beer.propTypes = {
  beer: PropTypes.shape({
    name: PropTypes.string.isRequired,
    tagline: PropTypes.string.isRequired,
    abv: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image_url: PropTypes.string.isRequired,
    food_pairing: PropTypes.array.isRequired,
  }).isRequired,
  handleOnClick: PropTypes.isRequired,
};

export default Beer;
