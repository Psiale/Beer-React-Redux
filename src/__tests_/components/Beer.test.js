import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Beer from '../../components/Beer';

configure({ adapter: new Adapter() });

const beer = {
  id: 1,
  name: 'Beer',
  tagline: 'Supe',
  abv: 18,
  imageUrl: 'Something',
};

const component = <Beer beer={beer} handleOnClick={() => {}} category="Pale Ale" />;
it('renders correctly', () => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('finds the category text', () => {
  const wrapper = shallow(<Beer beer={beer} handleOnClick={() => {}} category="Pale Ale" />);
  const divs = wrapper.find('div');
  expect(divs).toHaveLength(5);
});
