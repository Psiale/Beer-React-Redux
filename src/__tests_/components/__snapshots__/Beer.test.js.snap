// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-hidden="true"
  className="mainContainer"
  onClick={[Function]}
  onKeyPress={[Function]}
>
  <div
    className="category"
    style={
      Object {
        "backgroundColor": "rgb(76, 246, 224)",
      }
    }
  >
    Pale Ale
  </div>
  <img
    alt=""
    className="image"
    src="Something"
  />
  <div
    className="name"
  >
    Beer
  </div>
  <div
    className="abv"
  >
    18
    ABV
  </div>
  <div
    className="tagline"
  >
    <span>
      Supe
    </span>
  </div>
</div>
`;
